server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java

```diff
@@ -0,0 +1,114 @@
+        ServerConnector sc = (ServerConnector) jetty.getConnectors()[0];
+        HTTP2CServerConnectionFactory factory =
+                new HTTP2CServerConnectionFactory(new HttpConfiguration());
+        factory.setRateControlFactory(new RateControl.Factory() {});
+        sc.addConnectionFactory(factory);
+
+        final WebAppContext context =
+                new WebAppContext(null, "/", null, null, null, new ErrorPageErrorHandler(), SESSIONS);
+        try {
+            String knownFile = "/ide/index.html";
+            URL ide = JettyBackedGrpcServer.class.getResource(knownFile);
+            context.setBaseResource(Resource.newResource(ide.toExternalForm().replace("!" + knownFile, "!/")));
+        } catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+
+        // context.addFilter(NoCacheFilter.class, "*.nocache.js", EnumSet.noneOf(DispatcherType.class));
+        // context.addFilter(CacheFilter.class, "*.cache.js", EnumSet.noneOf(DispatcherType.class));
+        //
+        // // For the Web UI, cache everything in the static folder
+        // // https://create-react-app.dev/docs/production-build/#static-file-caching
+        // context.addFilter(NoCacheFilter.class, "/iriside/*", EnumSet.noneOf(DispatcherType.class));
+        // context.addFilter(CacheFilter.class, "/iriside/static/*", EnumSet.noneOf(DispatcherType.class));
+
+        // Always add eTags
+        context.setInitParameter("org.eclipse.jetty.servlet.Default.etags", "true");
+        context.setSecurityHandler(new ConstraintSecurityHandler());
+
+        context.setContextPath("/");
+        context.addFilter(new FilterHolder(filter), "/*", EnumSet.noneOf(DispatcherType.class));
+        jetty.setHandler(context);
+
+        // set up websocket for grpc-web
+        JakartaWebSocketServletContainerInitializer.configure(context, (servletContext, container) -> {
+            container.addEndpoint(
+                    ServerEndpointConfig.Builder.create(WebSocketServerStream.class, "/{service}/{method}")
+                            .configurator(new ServerEndpointConfig.Configurator() {
+                                @Override
+                                public <T> T getEndpointInstance(Class<T> endpointClass) throws InstantiationException {
+                                    return (T) filter.create(WebSocketServerStream::new);
+                                }
+                            })
+                            .build());
+        });
+    }
+
+    @Override
+    public void start() throws IOException {
+        try {
+            jetty.start();
+        } catch (RuntimeException exception) {
+            throw exception;
+        } catch (Exception exception) {
+            throw new IOException(exception);
+        }
+    }
+
+    @Override
+    public void shutdown() throws Exception {
+        jetty.stop();
+    }
+
+    @Override
+    public void shutdownNow() {
+        // jetty.stop();
```

Comment:
Not applicable?

Reply:
This is fixed now, the whole GrpcServer interface and impls have been rewritten, doc'd.